src/fsharp/MethodCalls.fs

```diff
@@ -43,13 +43,16 @@ open FSharp.Compiler.ExtensionTyping
 ///
 /// The bool indicates if named using a '?' 
 type CallerArg<'T> = 
-    /// CallerArg(ty, range, isOpt, exprInfo)
-    | CallerArg of TType * range * bool * 'T  
-    member x.Type = (let (CallerArg(ty, _, _, _)) = x in ty)
-    member x.Range = (let (CallerArg(_, m, _, _)) = x in m)
-    member x.IsOptional = (let (CallerArg(_, _, isOpt, _)) = x in isOpt)
-    member x.Expr = (let (CallerArg(_, _, _, expr)) = x in expr)
+    { Type: TType 
+      Range: range
+      IsOptional: bool 
+      Expr: 'T }
     
+module CallerArg =
```

Comment:
Please remove these functions or change to static `Create` members, we basically never use this style of "extra module for constructors of a record type" in the compiler